foxtrot-common/src/main/java/com/flipkart/foxtrot/common/query/numeric/NumericBinaryFilter.java

```diff
@@ -61,4 +59,37 @@ public boolean isFilterTemporal() {
         }
         return validationErrors;
     }
+
+    @Override
+    public int hashCode() {
+        int result = getOperator().hashCode();
+        result = 31 * result + getField().hashCode();
+        if(!getField().equals("_timestamp")) {
+            result = result * 21 + (getValue() == null
+                                    ? 43
+                                    : getValue().hashCode());
+        }
+        else {
+            result = result * 21 + Long.valueOf(getValue().longValue() / (long) 30000).hashCode();
```

Comment:
Check comments. It is not correct to do this indiscriminately. Create and abstract function to get timestamp hash code.